pip/index.py

```diff
@@ -412,10 +413,17 @@ def _find_all_versions(self, project_name):
         for location in url_locations:
             logger.debug('* %s', location)
 
+        formats = set(["source"])
+        if self.use_wheel:
+            formats.add("binary")
+        search = Search(
+            project_name.lower(),
+            pkg_resources.safe_name(project_name.lower()).lower(),
```

Comment:
I think ideally, we should have a `def canonical_name` somewhere in pip that would do this.
And given what `safe_name` does, one of the `lower` is redundant (and useless).